fixes #5814

mighty-phoenix opened the pull request:
Removed the node that caused the appearance of two edit buttons in exploration player.

<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->

## Checklist
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The linter/Karma presubmit checks have passed.
  - These should run automatically, but if not, you can manually trigger them locally using `python scripts/pre_commit_linter.py` and `bash scripts/run_frontend_tests.sh`.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR follows the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide).
- [ ] The PR is assigned to an appropriate reviewer.
  - If you're a new contributor, please ask on [Gitter](https://gitter.im/oppia/oppia-chat) for someone to assign a reviewer.
  - If you're not sure who the appropriate reviewer is, please assign to the issue's "owner" -- see the "talk-to" label on the issue.

mighty-phoenix committed:
fixes #5814

Removed the node that caused the appearance of two edit buttons in exploration player.

vojtechjelinek commented:
Thanks, but did you try to test this? When you're logged in and looking at exploration made by another person the navbar looks like this:
![image](https://user-images.githubusercontent.com/10142938/47885058-8a500b00-de33-11e8-8e9d-d723d37a38ff.png)

Also you can't really remove the second edit button because it has different function and is supposed to show for suggesting changes to exploration.

vojtechjelinek commented on core/templates/dev/head/pages/exploration_player/learner_local_nav.html:
```diff
@@ -11,7 +11,6 @@
-    <a ng-if="!canEdit && username" class="protractor-test-exploration-suggestion-popup-link" ng-click="showLearnerSuggestionModal()" uib-tooltip="Suggest Changes" tooltip-placement="bottom" target="_blank">
       <i class="material-icons">&#xE254;</i>
       <span class="oppia-icon-accessibility-label">Suggest Changes</span>
     </a>
```
`</a>` without starting `<a>`

mighty-phoenix committed:
fixes #5814

Replaced the second edit icon with a more appropriate icon for suggesting changes in exploration, and placed it properly in the navbar.

rakshitkumarcse commented:
@vojtechjelinek Thanks for pointing that out. Sorry for the mess. I am contributing for the first time. But I will make sure that I test the changes before making a PR, and also understand the whole situation. I am a fast learner. With your guidance, I shall become a good contributor and hopefully help in improving Oppia.
I have made a second commit. Kindly review.

mighty-phoenix reviewed (commented):
.

mighty-phoenix commented:
@vojtechjelinek Thanks for pointing that out. Sorry for the mess. I am contributing for the first time. But I will make sure that I test the changes before making a PR, and also understand the whole situation the next time. I am a fast learner. With your guidance, I shall become a good contributor and hopefully help in improving Oppia.
I have made a second commit. Kindly review and merge.

seanlip commented:
@rakshitkumarcse Could you please confirm that you have tested this manually on your local dev server in the various possible cases, and explain exactly which tests you've tried?

mighty-phoenix commented:
@seanlip Ya I have tested it manually on my local dev server. There is only one possible case to encounter that situation as specified by @vojtechjelinek in the issue description. I have positively tested it.

seanlip commented:
Take a careful look at the code. There are multiple cases based on what is described in the ng-if statements. You should test each of these.

mighty-phoenix commented:
@seanlip Yes. I have done tests for those cases. It's working perfectly. Can assure confidently.

seanlip commented:
Thanks. Please specify in detail which cases you have tested, as I asked in my original comment.

mighty-phoenix commented:
@seanlip 
Case where:
1. The author of exploration himself opens the exploration player,
2. Another user, who has not written the exploration opens the exploration player.
3. When sign in is not done and the user opens the exploration player.

I have also ensured that all the functions involved are working correctly by manually making two different users and multiple explorations.

seanlip commented:
OK, thanks for confirming @rakshitkumarcse! In that case, passing the review back to @vojtechjelinek.

seanlip commented on core/templates/dev/head/pages/exploration_player/learner_local_nav.html:
```diff
@@ -11,8 +11,10 @@
+  <li>
     <a ng-if="!canEdit && username" class="protractor-test-exploration-suggestion-popup-link" ng-click="showLearnerSuggestionModal()" uib-tooltip="Suggest Changes" tooltip-placement="bottom" target="_blank">
-      <i class="material-icons">&#xE254;</i>
+      <i class="material-icons">contact_support</i>
```
Please could you explain why this change was needed? It breaks convention with the other uses of `<i class="material-icons">` in the app.

If it's not needed, please revert it. Thanks!

mighty-phoenix replied:
This change was needed because earlier, 2 identical icons appeared. The one that is now there for "edit" was used twice. Hence the change.

seanlip replied:
That seems incorrect to me. I didn't think this PR was supposed to introduce a new icon altogether.

However, I'll defer to @vojtechjelinek for comment.

rakshitkumarcse replied:
@seanlip Even I thought so. But then in my last commit, @vojtechjelinek recommended me to do this as well.

mighty-phoenix replied:
https://github.com/oppia/oppia/issues/5814#issue-376549738
@seanlip The title and description of the issue clearly stated that only one edit icon should be there.
In my first commit, I had mistakenly removed the icon and didn't even replace it with anything. But then @vojtechjelinek made me realise that this icon/button had altogether a different role. It called a different function. So I ultimately replaced the original edit icon being used for "suggest change" with the most appropriate replacement.

mighty-phoenix commented:
@seanlip  Thanks for the guidance. I will fix issues in a more organized way from the next time. Actually, I am new to open source. And I do not have much prior experience in open source. I am a sophomore pursuing Computer Science. I am experienced in web dev, python and json. I hope my work improves. I found Oppia in the list of GSOC organizations and found it to match my interests the most. Thanks for bearing with me. I will adapt to this work ASAP.

mighty-phoenix commented:
@vojtechjelinek Kindly review and merge.

vojtechjelinek commented:
Hi, maybe I phrased the problem wrongly. The problem **is not** that there are two buttons with the same icon, the problem is that one of these buttons shows up and then disappears. The same thing happens with your solution. When the page is loading the suggestion button shows up for like 1 second an then dissappears. 
![image](https://user-images.githubusercontent.com/10142938/47950336-1146d580-df51-11e8-9d5c-61ef6a1013af.png)
The thing we want to achieve is that the button that should be hidden will never show and then hide. The fix to this problem would probably require some modifications to Javascrpt also, but I'm not sure.
If you still don't know what exactly is needed just ask me.

mighty-phoenix commented:
@vojtechjelinek It's well understood now. I'll fix it and make a PR. Also now that I have changed one of the icons, is it fine if it stays changed? I feel it is more appropriate for "suggesting changes". @seanlip What do you think?

mighty-phoenix committed:
FIxes #5814

mighty-phoenix commented:
@vojtechjelinek I have changed the icon back to the original one. Also, the issue is fixed.
Can you please make the final review and merge?

apb7 commented:
Just a note here -- @rakshitkumarcse, please do not use GitHub's online editor. This skips all the pre-submit checks and makes your PR more prone to lint errors. Thanks!

mighty-phoenix commented:
@apb7 Sure. Wasn't aware of that. lint error won't occur since I directly copied and pasted my code into the editor. But I will make sure I don't use editor from the next time since it skips all pre-submit checks.
Thanks!

vojtechjelinek reviewed (approved):
Thanks! LGTM! I didn't know something like `ng-cloak` existed so god job.

vojtechjelinek commented:
I only discovered another bug (it's not related to you fix because it happens in production too) that the accessibility tooltips are missplaced for some buttons:
![image](https://user-images.githubusercontent.com/10142938/47950981-e6fa1580-df5a-11e8-82f0-56f63b167edf.png)
For some buttons they're correct:
![image](https://user-images.githubusercontent.com/10142938/47950993-fda06c80-df5a-11e8-902d-e8f3829c9225.png)
This might be related to the fact that we are hiding some buttons.

@rakshitkumarcse Would you create GitHub issue for this bug and if you want try to fix it?

mighty-phoenix commented:
@vojtechjelinek Sure. Will do it a few hours later and fix it as well.

mighty-phoenix commented:
@vojtechjelinek I opened the issue #5821 .
@seanlip Kindly merge and close this issue.
